pkg/funcs/functions.go

```diff
@@ -0,0 +1,100 @@
+package pkg
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+	"time"
+
+	"github.com/araddon/dateparse"
+	oc "github.com/cloudprivacylabs/opencypher"
+	"github.com/neo4j/neo4j-go-driver/neo4j"
+)
+
+func init() {
+	oc.RegisterGlobalFunc(oc.Function{
+		Name:      "age",
+		MinArgs:   0,
+		MaxArgs:   2,
+		ValueFunc: ageFunc,
+	})
+}
+
+func ageFunc(ctx *oc.EvalContext, args []oc.Value) (oc.Value, error) {
+	if args[0].Get() == nil {
+		return oc.RValue{}, nil
+	}
+	var n4jDate neo4j.Date
+	var n4jDateTime neo4j.LocalDateTime
+	var err error
+	// check if argument a neo4j.Date or neo4j.LocalDateTime
+	if n4jDate, err = valueAsNeo4jDate(args[0]); err != nil {
```

Comment:
Can't do else here. The first error will terminate. If the input is neo4jdatetime, it will never check it.